### PR TITLE
Added tensor input support for ONNX and gluon flavors.

### PR DESCRIPTION
**Pull request by tomasatdatabricks**
## What changes are proposed in this pull request?
Add support for tensor input to the ONNX and gluon model flavors. This PR aims to preserve existing behavior when the model is invoked with DataFrame input. 

**ONNX flavor:**
 - support a single numpy array input
    Single numpy array is assumed to be a tensor and is supported iff the underlying model has exactly one input. 
    The array is passed to the model as a dictionary with a single item, the keys is the input name stored with the model and the value is the numpy array itself.
 - support dictionary input
   the input dictionary is passed to the model as is.
 - apply existing casting of float64 -> float32 to the new types.
   The existing model flavor checks if there are inputs that are expected to be float32 and were passed as float64. This PR preserves this behavior and applies it to all supported input types.
 
**Gluon flavor:** It is not clear to me if gluon supports dictionary or list of tensors as the input. For now, the flavor will only support a single numpy array (tensor). 

## How is this patch tested?
Added unit tests.

## Release Notes
The python function models for ONNX and gluon flavors now support tensor input. Onnx models can be called with tensors passed in as a single numpy array or a dictionary of numpy arrays. The gluon models can be called with tensors passed in as a single numpy array.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/onnx.py [hunk 1]
```diff
@@ -160,17 +160,18 @@ def __init__(self, path):
...
+        for input_name, input_type in self.inputs:
+            if input_type == "tensor(float)":
+                feed = feeds.get(input_name)
+                if feed is not None and feed.dtype == np.float64:
+                    feeds[input_name] = feed.astype(np.float32)
+        return feeds
 
     @experimental
-    def predict(self, dataframe):
+    def predict(self, data):
```
Comment (arjundc-db) [inline mlflow/onnx.py:172]: Please modify the documentation to reflect code change.

**Review by arjundc-db — COMMENTED** [review]
Thank you for adding this, I had a few comments.
It lgtm once those are addressed.

File: mlflow/gluon.py [hunk 2]
```diff
@@ -85,8 +86,14 @@ def predict(self, df):
...
         import mxnet as mx
 
-        ndarray = mx.nd.array(df.values)
-        return pd.DataFrame(self.gluon_model(ndarray).asnumpy())
+        if isinstance(data, pd.DataFrame):
+            ndarray = mx.nd.array(data.values)
+            return pd.DataFrame(self.gluon_model(ndarray).asnumpy())
+        elif isinstance(data, np.ndarray):
+            ndarray = mx.nd.array(data)
+            return self.gluon_model(ndarray).asnumpy()
```
[the remark this answers is not in the document]
Comment (tomasatdatabricks) [reply mlflow/gluon.py:94]: Good catch!

**Review by arjundc-db — APPROVED** [review]
lgtm, thanks @tomasatdatabricks

**Pull request merged**